bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -820,10 +820,12 @@ private void swapWriteCache() {
     }
 
     @Override
-    public void flush() throws IOException {
+    public void flush(boolean doCheckpointComplete) throws IOException {
         Checkpoint cp = checkpointSource.newCheckpoint();
         checkpoint(cp);
-        checkpointSource.checkpointComplete(cp, true);
+        if (doCheckpointComplete) {
```

Comment:
It is a little strange that we flush the WriteCache data into OS PageCache without writing the checkpoint `lastMark` into ledger directory.

Reply:
In SyncThread,  We can disableCheckpoint, just return without executing `checkpointSource.checkpointComplete(checkpoint, false);` I don't think this is strange.